tests/unit/worker/test_submission_worker.py

```diff
@@ -109,3 +113,43 @@ def test_get_or_create_sqs_queue_for_non_existing_queue(self):
         queue_url = self.sqs_client.get_queue_url(QueueName='test_queue_2')['QueueUrl']
         self.assertTrue(queue_url)
         self.sqs_client.delete_queue(QueueUrl=queue_url)
+
+
+class TestDownloadAndExtractFile(APITestCase):
+    def setUp(self):
+        super(TestDownloadAndExtractFile, self).setUp()
+
+        self.base_dir = "{}/JdyW0ZCFLr".format(tempfile.gettempdir())
+        self.download_filename = "VwYOaVWkdw"
+        self.download_path = "{}/{}".format(self.base_dir, self.download_filename)
+        self.extract_dirname = "gMTblaJFbM"
+        self.extract_path = "{}/{}".format(self.base_dir, self.extract_dirname)
+        self.correct_md5 = "86629294bdc8256500ca37a9afecbd79"
+        self.url = "https://file-examples.com/wp-content/uploads/2017/02/zip_5MB.zip"
+        create_dir(self.base_dir)
+
+    def tearDown(self):
+        shutil.rmtree(self.base_dir)
+
+    def test_download_and_extract_file_with_correct_url(self):
+        assert os.path.exists(self.base_dir)
+
+        download_and_extract_zip_file(self.url, self.download_path, self.extract_path)
+        # with open(self.download_path, "r") as f:
```

Comment:
@grave-panda can you remove the unnecessary code

Reply:
Yeah sorry about that, I forgot to cleanup

Reply:
Sorry about that, I forgot to clean up